### ISSUE TITLE
Allow using groups for shared VPC access

### ISSUE THREAD

**Issue by g-prkr** [terraform-google-modules/terraform-google-project-factory#814]
### TL;DR

Currently, the project creation creates new service accounts and updates the access for the shared VPC project by updating the IAM policy. This runs into limitations of the no. of principals that IAM policies support (1500), documented here - https://cloud.google.com/iam/docs/reference/rest/v1/Policy#:~:text=The%20bindings%20in,the%20Policy.

The recommended approach to fix this is to use a group which has access to the shared VPC project and adding the new SA to the group, instead of updating the policy. Making these configurable options will help large customers scale the automation more easily.

**Comment by g-awmalik** [comment]
I'm assuming the IAM policy you're mentioning is for the project service account on the shared VPC subnets as `roles/compute.networkUser`. This is happening [here](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/core_project_factory/main.tf#L204).

The customer has the option to set [`create_project_sa`](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/core_project_factory/variables.tf#L86) as false and instead set [`manage_group`](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/core_project_factory/variables.tf#L35) and provide the gsuite group as [`group_email`](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/core_project_factory/variables.tf#L17).

This was a service account won't be created and won't added to the shared VPC project's IAM policies while the group will to which access can be managed as needed.

Does this make sense to you?

**Comment by g-prkr** [comment]
Thanks.
The expectation is that the SA gets created for the new project and gets access to the shared VPC. In order to get access, instead of modifying the IAM policy, will the new SA gets added to the configured group.
Will setting create_project_sa to true and manage_group as well with the group_email achieve this?

**Comment by g-awmalik** [comment]
No it won't. It'll give both the SA and the group the networkUser role on the host project.

For this we'll need to introduce another flag that specifies if the created project SA should explicitly be given permissions or just added to the group.
